src/traces/parcats/parcats.js

```diff
@@ -0,0 +1,2086 @@
+            })
+        .each(function(d) {
+            // Update attriubutes of <tspan> elements
+            var newX;
+            var newAnchor;
+            if(catInRightDim(d)) {
+                // Place label to the right of category
+                newX = d.width + 5;
+                newAnchor = 'start';
+            } else {
+                // Place label to the left of category
+                newX = -5;
+                newAnchor = 'end';
+            }
+            d3.select(this)
+                .selectAll('tspan')
+                .attr('x', newX)
+                .attr('text-anchor', newAnchor);
+        });
+
+    // Update bands
+    // Initialize color band rects
+    var bandSelection = categorySelection
+        .selectAll('rect.bandrect')
+        .data(
+            /** @param {CategoryViewModel} catViewModel*/
+            function(catViewModel) {
+                return catViewModel.bands;
+            }, key);
+
+    var bandsSelectionEnter = bandSelection.enter()
+        .append('rect')
+        .attr('class', 'bandrect')
+        .attr('cursor', 'move')
+        .attr('stroke-opacity', 0)
+        .attr('fill', function(d) {
+            return d.color;
+        })
+        .attr('fill-opacity', 0);
+
+    bandSelection
+        .attr('fill', function(d) {
+            return d.color;
+        })
+        .attr('width', function(d) {
+            return d.width;
+        })
+        .attr('height', function(d) {
+            return d.height;
+        })
+        .attr('y', function(d) {
+            return d.y;
+        });
+
+    styleBandsNoHover(bandsSelectionEnter);
+
+    // Raise bands to the top
+    bandSelection.each(function() {Lib.raiseToTop(this);});
+
+    // Remove unused bands
+    bandSelection.exit().remove();
+}
+
+/**
+ * Create a ParcatsViewModel traces
+ * @param {Object} graphDiv
+ *  Top-level graph div element
+ * @param {Layout} layout
+ *  SVG layout object
+ * @param {Array.<ParcatsModel>} wrappedParcatsModel
+ *  Wrapped ParcatsModel for this trace
+ * @return {ParcatsViewModel}
+ */
+function createParcatsViewModel(graphDiv, layout, wrappedParcatsModel) {
+    // Unwrap model
+    var parcatsModel = wrappedParcatsModel[0];
+
+    // Compute margin
+    var margin = layout.margin || {l: 80, r: 80, t: 100, b: 80};
+
+    // Compute pixel position/extents
+    var trace = parcatsModel.trace,
+        domain = trace.domain,
+        figureWidth = layout.width,
+        figureHeight = layout.height,
+        traceWidth = Math.floor(figureWidth * (domain.x[1] - domain.x[0])),
+        traceHeight = Math.floor(figureHeight * (domain.y[1] - domain.y[0])),
+        traceX = domain.x[0] * figureWidth + margin.l,
+        traceY = layout.height - domain.y[1] * layout.height + margin.t;
+
+    // Handle path shape
+    // -----------------
+    var pathShape;
+    if(trace.line && trace.line.shape) {
+        pathShape = trace.line.shape;
+    } else {
+        pathShape = 'linear';
+    }
```

Comment:
yeah this kind of logic should be already taken care of by `supplyDefaults` so you can just say `pathShape = trace.line.shape`

Reply:
Done in eba1b4a